packages/compass-e2e-tests/helpers/commands/disconnect.ts

```diff
@@ -13,29 +13,42 @@ export async function disconnect(browser: CompassBrowser): Promise<void> {
     return;
   }
 
-  const cancelConnectionButtonElement = await browser.$(
-    Selectors.CancelConnectionButton
-  );
-  // If we are still connecting, let's try cancelling the connection first
-  if (await cancelConnectionButtonElement.isDisplayed()) {
-    try {
-      await browser.closeConnectModal();
-    } catch (e) {
-      // If that failed, the button was probably gone before we managed to
-      // click it. Let's go through the whole disconnecting flow now
+  if (!TEST_MULTIPLE_CONNECTIONS) {
+    const cancelConnectionButtonElement = await browser.$(
+      Selectors.CancelConnectionButton
+    );
+    // If we are still connecting, let's try cancelling the connection first
+    if (await cancelConnectionButtonElement.isDisplayed()) {
+      try {
+        await browser.closeConnectModal();
+      } catch (e) {
+        // If that failed, the button was probably gone before we managed to
+        // click it. Let's go through the whole disconnecting flow now
+      }
     }
-  }
 
-  await delay(100);
+    await delay(100);
+  }
 
   await browser.execute(() => {
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     require('electron').ipcRenderer.emit('app:disconnect');
   });
 
-  const element = await browser.$(Selectors.ConnectSection);
-  await element.waitForDisplayed();
+  if (TEST_MULTIPLE_CONNECTIONS) {
+    // For multiple connections we're making the assumption that there should be
+    // no active connections left. Use a different command if you expect to
+    // disconnect just one connection and still keep others around.
+    await browser.$(Selectors.SidebarTreeItems).waitForExist({ reverse: true });
```

Comment:
we'll need a different logic once the lists are merged, perhaps we should add another TODO here?

Reply:
I'm going to just merge in main once that's merged and then deal with the fallout before merging this PR.